action/package.json

```diff
@@ -11,21 +11,21 @@
     "test": "./test/bin/run-tests.sh --colors"
   },
   "devDependencies": {
-    "@types/git-url-parse": "^9.0.1",
-    "@types/jest": "^27.4.0",
-    "@types/node": "^17.0.16",
-    "@zeit/ncc": "^0.22.3",
-    "dotenv": "^16.0.0",
-    "git-url-parse": "^13.1.0",
-    "jest": "^27.5.1",
-    "prettier": "^2.3.0",
-    "ts-jest": "^27.1.3",
-    "ts-node": "^10.5.0",
-    "typescript": "^4.5.5"
+    "@types/git-url-parse": "^9.0.3",
+    "@types/jest": "^29.5.12",
+    "@types/node": "^20.11.16",
+    "@vercel/ncc": "^0.38.1",
+    "dotenv": "^16.4.1",
+    "git-url-parse": "^14.0.0",
+    "jest": "^29.7.0",
+    "prettier": "^3.2.5",
+    "ts-jest": "^29.1.2",
+    "ts-node": "^10.9.2",
+    "typescript": "^5.3.3"
```

Comment:
There are some build and test failures with the latest version of TypeScript, could you look into these?